src/base/solvers/configEnvVars.m

```diff
@@ -186,7 +186,8 @@ function configEnvVars(printLevel)
 
         % check for 32-bit
         if ~isempty(strfind(computer('arch'), '64'))
-            subDir = [filesep, osPath, filesep, 'matlab'];
+            subDir = [filesep, 'matlab'];
+            %subDir = [filesep, osPath, filesep, 'matlab'];
         end
```

Comment:
What is your installation path to `GUROBI`? If you do not have a standard path, did you set the environment variables as explained [here](https://github.com/opencobra/cobratoolbox/blob/master/docs/source/installation/solvers.md#gurobi)? We can continue adding specific paths to the [list](https://github.com/opencobra/cobratoolbox/blob/master/src/base/solvers/configEnvVars.m), but that is not ideal (hence the environment variables).

I will have a look at where to improve the documentation.

`GUROBI` (as well as `ILOG CPLEX`, `TOMLAB CPLEX` and `MOSEK`) all must be removed when the interface is changed to `matlab`. That is actually the purpose of that code. The reason is that all these solvers shade the default `MATLAB` functions (such as `linprog.m`). Without removing the path of these solvers explicitly, the interface `matlab` cannot be used if the user has another solver installed. The path to the solver is removed, but there is also the part of the code that re-adds the path should then the solver be requested. 

This has been tested multiple times, and worked without any issues. What exactly was the error message thrown on your machine that made you change the lines?